### PR TITLE
TST: use env variable instead of pytest option for testing ArrayManager

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
I noticed a problem with the current `--array-manager` pytest option to test the ArrayManager, while checking the window tests. 

The problem with the option is that `conftest.py` files are read first, *before* handling the option in `pytest_sessionstart`. This means that any DataFrame or Series that gets constructed on *import* of a `conftest.py` file, has already been created before the setting is changed, and will thus be created with the wrong manager. 
Example of this is a DataFrame that is not created *inside* a (fixture) function, but created top-level in the conftest.py file (or eg used in parametrization). 

Luckily we don't have too many of those, but one example in window/conftest.py:

https://github.com/pandas-dev/pandas/blob/74a68980d116c843d04b3ccb6ab9368f19637d9b/pandas/tests/window/conftest.py#L352-L367

or one example in the toplevel conftest.py:

https://github.com/pandas-dev/pandas/blob/74a68980d116c843d04b3ccb6ab9368f19637d9b/pandas/conftest.py#L597-L608

So to avoid needing to rewrite those (and continue to ensure such cases don't get added), I don't see another way as using the environment variable approach instead of the pytest option (although the option is much easier to use interactively ..)

cc @jreback

### CODE REVIEW

File: pandas/core/config_init.py [hunk 1]
```diff
@@ -478,14 +479,19 @@ def use_inf_as_na_cb(key):
     _use_inf_as_na(key)
 
 
+# Get the default from an environment variable, if set, otherwise defaults to "block"
```
Comment (jreback) [inline pandas/core/config_init.py:482]: i would just inline this on L494 (and add a 1-liner comment before) and/or update the option description with the text on how to control this

**Review by jreback — CHANGES_REQUESTED** [review]
small comment, looks fine to me

**Comment by jreback** [comment]
> Luckily we don't have too many of those, but one example in window/conftest.py:

should we just change / check for these?

**Comment by jorisvandenbossche** [comment]
> should we just change / check for these?

I think that's a bit fragile to ensure that in the future (and in theory there is also nothing wrong with it, just not working with this option that influences the dataframe creation). Would there be an easy way to check this?

**Comment by jreback** [comment]
> > should we just change / check for these?
> 
> I think that's a bit fragile to ensure that in the future (and in theory there is also nothing wrong with it, just not working with this option that influences the dataframe creation). Would there be an easy way to check this?

agree its fragile, but would be good to check it, yeah not sure its easy.

**Review by jreback — APPROVED** [review]
lgtm

**Comment by jorisvandenbossche** [comment]
If someone comes up with a good approach for keeping it with the pytest option, we can always change back. But for now I want to ensure I am correctly testing ArrayManager ;)

**Pull request merged**